papertty/drivers/drivers_full.py

```diff
@@ -270,6 +270,177 @@ def sleep(self):
         self.send_data(0xa5)
 
 
+class EPD3in7(WaveshareFull):
+    """Waveshare 3.7" - monochrome"""
+
+    def __init__(self):
+        super().__init__(name='3.7" BW', width=280, height=480)
+
+    # not ported: 4-greyscale mode, partial refresh (~600-900 ms, EPD_3in7.c)
+
+    lut_1Gray_GC  = [
+        0x2A,0x05,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x05,0x2A,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x2A,0x15,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x05,0x0A,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x02,0x03,0x0A,0x00,0x02,0x06,0x0A,0x05,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x22,0x22,0x22,0x22,0x22
+    ]
+
+    lut_1Gray_DU  = [
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x01,0x2A,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x0A,0x55,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x05,0x05,0x00,0x05,0x03,0x05,0x05,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x22,0x22,0x22,0x22,0x22
+    ]
+
+    lut_1Gray_A2  = [
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x0A,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x05,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00, 
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00, 
+        0x00,0x00,0x03,0x05,0x00,0x00,0x00,0x00,0x00,0x00, 
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00, 
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00, 
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00, 
+        0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00, 
+        0x22,0x22,0x22,0x22,0x22
+    ]
+
+    def ReadBusy(self):
```

Comment:
For consistency, should rename this method (note to self - maybe refactor base `wait_until_idle` to take parameters to cater for all variants so it doesn't need to be overridden):

```suggestion
    def wait_until_idle(self):
```